File: obra-do-berco-gestante/src/main/java/com/github/luuispnp/obradoberco/gestante/entity/Gestante.java
package com.github.luuispnp.obradoberco.gestante.entity;

import com.github.luuispnp.obradoberco.gestante.enums.EstadoCivil;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.UUID;

@Getter
@Setter
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Entity
@Table(name = "tb_gestante")
public class Gestante {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column
    @EqualsAndHashCode.Include
    private UUID id;

    @Column(nullable = false, length = 150)
    private String nome;

    @Column(nullable = false, unique = true, length = 11)
    private String cpf;

    @Column(name = "numero_identidade", unique = true, length = 25)
    private String numeroIdentidade;

    @Column(name = "data_nascimento", nullable = false)
    private LocalDate dataNascimento;

    @Embedded
    private Endereco endereco;

    @Enumerated(EnumType.STRING)
    @Column(name = "estado_civil", nullable = false, length = 20)
    private EstadoCivil estadoCivil;

    @Column(nullable = false)
    private String telefone;

    @Column(unique = true)
    private String email;

    @Column(name = "data_cadastro", nullable = false)
    private LocalDate dataCadastro;

    @Column(nullable = false)
    private Boolean ativo;

    @PrePersist
    public void prePersist() {
        if (dataCadastro == null)
            dataCadastro = LocalDate.now();

        ativo = true;
    }

}

File: obra-do-berco-frontend/src/types/gestante.ts
export type EstadoCivil = "SOLTEIRA" | "CASADA" | "UNIAO_ESTAVEL" | "VIUVA";

export interface Endereco {
  logradouro: string;
  bairro: string;
  cidade: string;
  referencia?: string | null;
}

export interface EnderecoInput {
  cep: string;
  logradouro: string;
  bairro: string;
  cidade: string;
  referencia?: string;
}

export interface Gestante {
  id: string;
  nome: string;
  cpf: string;
  numeroIdentidade?: string | null;
  dataNascimento: string;
  endereco: Endereco;
  estadoCivil: EstadoCivil;
  telefone: string;
  email?: string | null;
  dataCadastro: string;
  ativo: boolean;
}

export interface GestanteInput {
  nome: string;
  cpf: string;
  numeroIdentidade?: string;
  dataNascimento: string;
  estadoCivil: EstadoCivil;
  telefone: string;
  email?: string;
  endereco: EnderecoInput;
}

export interface GestanteUpdateInput {
  nome: string;
  estadoCivil: EstadoCivil;
  telefone: string;
  email?: string;
  endereco: EnderecoInput;
}

export interface GestanteFiltro {
  nome?: string;
  cpf?: string;
  telefone?: string;
}
